release-lock-file-spec.md

```diff
@@ -0,0 +1,188 @@
+# Release Lock File Specification
+
+This document defines the specification for the **Release Lock File**.  The
+release lock file provides metadata about the package and in most cases should
+provide sufficient information about the packaged contracts and its
+dependencies to do bytecode verification of its contracts.
+
+
+## Guiding Principles
+
+The release lock file specification makes the following assumptions about the
+document lifecycle.
+
+1. Release lock files are intended to be generated programatically by package management software as part of the release process.
+2. Release lock files will be consumed by package managers during tasks like installing package dependencies or building and deploying new releases.
+3. Release lock files will typically **not** be stored alongside the source, but rather by package registries *or* referenced by package registries and stored in something akin to IPFS.
+
+
+## Format
+
+The canoonical format for the package manifest is a JSON document containing a
```

Comment:
* Def: **Canoonical:** *A humorous action performed in a canoe at or right around 12 Noon.*